x/auth/simulation/genesis.go

```diff
@@ -0,0 +1,136 @@
+	var maxMemoChars uint64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, MaxMemoChars, &maxMemoChars, simState.Rand,
+		func(r *rand.Rand) { maxMemoChars = GenMaxMemoChars(r) },
+	)
+
+	var txSigLimit uint64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, TxSigLimit, &txSigLimit, simState.Rand,
+		func(r *rand.Rand) { txSigLimit = GenTxSigLimit(r) },
+	)
+
+	var txSizeCostPerByte uint64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, TxSizeCostPerByte, &txSizeCostPerByte, simState.Rand,
+		func(r *rand.Rand) { txSizeCostPerByte = GenTxSizeCostPerByte(r) },
+	)
+
+	var sigVerifyCostED25519 uint64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, SigVerifyCostED25519, &sigVerifyCostED25519, simState.Rand,
+		func(r *rand.Rand) { sigVerifyCostED25519 = GenSigVerifyCostED25519(r) },
+	)
+
+	var sigVerifyCostSECP256K1 uint64
+	simState.AppParams.GetOrGenerate(
+		simState.Cdc, SigVerifyCostSECP256K1, &sigVerifyCostSECP256K1, simState.Rand,
+		func(r *rand.Rand) { sigVerifyCostSECP256K1 = GenSigVerifyCostSECP256K1(r) },
+	)
+
+	params := types.NewParams(maxMemoChars, txSigLimit, txSizeCostPerByte,
+		sigVerifyCostED25519, sigVerifyCostSECP256K1)
+	genesisAccs := RandomGenesisAccounts(simState)
+
+	authGenesis := types.NewGenesisState(params, genesisAccs)
+
+	fmt.Printf("Selected randomly generated auth parameters:\n%s\n", codec.MustMarshalJSONIndent(simState.Cdc, authGenesis.Params))
+	simState.GenState[types.ModuleName] = simState.Cdc.MustMarshalJSON(authGenesis)
+}
+
+// RandomGenesisAccounts returns randomly generated genesis accounts
+func RandomGenesisAccounts(simState *module.SimulationState) (genesisAccs exported.GenesisAccounts) {
+	for i, acc := range simState.Accounts {
+		coins := sdk.Coins{sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(simState.InitialStake))}
+		bacc := types.NewBaseAccountWithAddress(acc.Address)
+		if err := bacc.SetCoins(coins); err != nil {
+			panic(err)
+		}
+
+		var gacc exported.GenesisAccount = &bacc
+
+		// Only consider making a vesting account once the initial bonded validator
+		// set is exhausted due to needing to track DelegatedVesting.
+		if int64(i) > simState.NumBonded && simState.Rand.Intn(100) < 50 {
+			var endTime int64
+
+			startTime := simState.GenTimestamp.Unix()
+
+			// Allow for some vesting accounts to vest very quickly while others very slowly.
+			if simState.Rand.Intn(100) < 50 {
+				endTime = int64(simulation.RandIntBetween(simState.Rand, int(startTime)+1, int(startTime+(60*60*24*30))))
+			} else {
+				endTime = int64(simulation.RandIntBetween(simState.Rand, int(startTime)+1, int(startTime+(60*60*12))))
+			}
+
+			if simState.Rand.Intn(100) < 50 {
+				if simState.Rand.Intn(100) < 50 {
+					gacc = vestingtypes.NewContinuousVestingAccount(&bacc, startTime, endTime)
+				} else {
+					gacc = vestingtypes.NewDelayedVestingAccount(&bacc, endTime)
+				}
+			} else {
+				gacc = vesting.NewManualVestingAccount(&bacc, nil, bacc.Address)
```

Comment:
Can we designate one unlocker instead of setting each account to be its own unlocker?

Reply:
I think this just make things easier to deal with. I think this is OK

Reply:
Yoongbok is right, it was meant to make things easier. It would be hard to track a fixed unlocker in simulation, unless we put it in genesis state and store like shield admin.